spring-pulsar/src/test/java/org/springframework/pulsar/core/reactive/ReactivePulsarTemplateTests.java

```diff
@@ -69,10 +72,17 @@ void sendMessageWithSpecificSchemaTest() throws Exception {
 						senderSpec);
 				ReactivePulsarSenderTemplate<Foo> pulsarTemplate = new ReactivePulsarSenderTemplate<>(producerFactory);
 				pulsarTemplate.setSchema(Schema.JSON(Foo.class));
-				Foo foo = new Foo("Foo-" + UUID.randomUUID(), "Bar-" + UUID.randomUUID());
-				pulsarTemplate.send(foo).subscribe();
-				assertThat(consumer.receiveAsync()).succeedsWithin(Duration.ofSeconds(3)).extracting(Message::getValue)
-						.isEqualTo(foo);
+
+				List<Foo> foos = new ArrayList<>();
+				for (int i = 0; i < 10; i++) {
+					foos.add(new Foo("Foo-" + UUID.randomUUID(), "Bar-" + UUID.randomUUID()));
+				}
+				pulsarTemplate.send(Flux.fromIterable(foos)).subscribe();
+
+				for (int i = 0; i < 10; i++) {
+					assertThat(consumer.receiveAsync()).succeedsWithin(Duration.ofSeconds(3))
```

Comment:
I do like that this also verifies the order contract. 

> @return the ids assigned by the broker to the published messages in the same order
	 * as they were sent

Does "sent" imply the order that I handed them to the API in my flux/publisher or the order they ended up being sent? I think this would almost always be true, but what provides this guarantee? What about failed message in the middle of the flux - and then it is retried maybe? Just playing devils' advocate.

Reply:
The reactive driver uses `flatMapSequential` on the `sendAsync` results. That will keep the order and should be tested in the reactive driver (not done atm).
I changed the logic when the template doesn't have a schema to also use `flatMapSequential`. I'll see if I can add a test to enforce the verification of the order.

Reply:
Failure of a mono in the flux will be handled as a usual r.s exception.
Maybe @lhotari can provide more details.